medmodels/medrecord/_overview.py

```diff
@@ -52,14 +50,17 @@ def extract_attribute_summary(
         Dict[MedRecordAttribute, Union[TemporalAttributeInfo, NumericAttributeInfo,
             StringAttributeInfo]: Summary of node or edge attributes.
     """
-    data = pl.DataFrame(data=[{"id": k, **v} for k, v in attribute_dictionary.items()])
+    list_of_dicts = [{**v} for v in attribute_dictionary.values()]
 
-    data_dict = {}
+    data = {}
+    for dictionary in list_of_dicts:
+        for key, value in dictionary.items():
+            data.setdefault(key, []).append(value)
 
-    attributes = sorted([col for col in data.columns if col != "id"])
+    data_dict = {}
 
-    for attribute in attributes:
-        attribute_values = data[attribute].drop_nulls()
+    for attribute in sorted(data.keys()):
+        attribute_values = [value for value in data[attribute] if value is not None]
 
         if len(attribute_values) == 0:
```

Comment:
Maybe it looks better this way:
```suggestion
        if not attribute_values:
```
@JabobKrauskopf ?

Reply:
imo `len(attribute_values) == 0` is more expressive